src/coreclr/jit/emitarm.cpp

```diff
@@ -7678,6 +7678,62 @@ void emitter::emitDispInsHelp(
     printf("\n");
 }
 
+/*****************************************************************************
+ *
+ *  Handles printing of LARGEJMP pseudo-instruction.
+ */
+
+void emitter::emitDispLargeJmp(
+    instrDesc* id, bool isNew, bool doffs, bool asmfm, unsigned offset, BYTE* code, size_t sz, insGroup* ig)
+{
+    // Note: don't touch the actual instrDesc. If we accidentally messed it up, it would create a very
+    // difficult to find bug.
+
+    instrDescJmp  idJmp;
+    instrDescJmp* pidJmp = &idJmp;
+
+    memset(&idJmp, 0, sizeof(idJmp));
+
+    pidJmp->idIns(emitJumpKindToIns(emitReverseJumpKind(emitInsToJumpKind(id->idIns())))); // reverse the
+                                                                                           // conditional
+                                                                                           // instruction
+    pidJmp->idInsFmt(IF_T1_K);
+    pidJmp->idInsSize(emitInsSize(IF_T1_K));
+    pidJmp->idjShort = 1;
+    pidJmp->idAddr()->iiaSetInstrCount(1);
+    pidJmp->idDebugOnlyInfo(id->idDebugOnlyInfo()); // share the idDebugOnlyInfo() field
+
+    size_t bcondSizeOrZero = (code == NULL) ? 0 : 2; // branch is 2 bytes
+    emitDispInsHelp(pidJmp, false, doffs, asmfm, offset, code, bcondSizeOrZero,
+                    NULL /* force display of pc-relative branch */);
+
+    code += bcondSizeOrZero;
+    offset += 2;
+
+    // Next, display the unconditional branch
+
+    // Reset the local instrDesc
+    memset(&idJmp, 0, sizeof(idJmp));
+
+    pidJmp->idIns(INS_b);
+    pidJmp->idInsFmt(IF_T2_J2);
+    pidJmp->idInsSize(emitInsSize(IF_T2_J2));
+    pidJmp->idjShort = 0;
```

Comment:
This is set to 1 above and then 0 here. Are both needed? Does pidJmp point to a different piece of memory in these cases?

Reply:
`idjShort` is initially set to 1 to satisfy an [assert ](https://github.com/dotnet/runtime/blob/058f83b9102aea36bab3ffe1068d2d842a07129d/src/coreclr/jit/emitarm.cpp#L7616) in `emitDispInsHelp` , as we emit a short conditional branch first for the pseudo-instruction. While the following unconditional branch is long (thus suggesting `idjShort` should be 0), I don't see any asserts in its control flow through `emitDispInsHelp` that require `idjShort == 0`, though that could change in the future.

This looks weird because `pidJmp` is used to emit both the short conditional and the long unconditional branches, but I think the logic is correct.

Reply:
The memory pointed to gets zeroed between uses (so zeroing isn't required, perhaps, but in this case it's best to be clear and have it).

Reply:
I agree, clarity is important. It might be worth a comment to explain that two different things are being done through the ptr but it's not a problem without.